packages/desktop-client/src/components/accounts/MobileAccountDetails.js

```diff
@@ -0,0 +1,180 @@
+import React, { useMemo } from 'react';
+import { Link } from 'react-router-dom';
+
+import { Label, View, Input, Button } from 'loot-design/src/components/common';
+import CellValue from 'loot-design/src/components/spreadsheet/CellValue';
+import Text from 'loot-design/src/components/Text';
+import { colors, mobileStyles as styles } from 'loot-design/src/style';
+import Add from 'loot-design/src/svg/v1/Add';
+import CheveronLeft from 'loot-design/src/svg/v1/CheveronLeft';
+import Search from 'loot-design/src/svg/v1/Search';
+
+import { TransactionList } from './MobileTransaction';
+
+class TransactionSearchInput extends React.Component {
+  state = { text: '' };
+
+  performSearch = () => {
+    this.props.onSearch(this.state.text);
+  };
+
+  onChange = text => {
+    this.setState({ text }, this.performSearch);
+  };
+
+  render() {
+    const { accountName } = this.props;
+    const { text } = this.state;
+
+    return (
+      <View
+        style={{
+          flexDirection: 'row',
+          alignItems: 'center',
+          backgroundColor: colors.n11,
+          margin: '11px auto 4px',
+          borderRadius: 4,
+          padding: 10,
+          width: '100%'
+        }}
+      >
+        <Search width="20" height="20" style={{ color: colors.n7 }} />
+        <Input
+          value={text}
+          onUpdate={this.onChange}
+          placeholder={`Search ${accountName}`}
+          style={{
+            backgroundColor: colors.n11,
+            border: `1px solid ${colors.n9}`,
+            fontSize: 15,
+            flex: 1,
+            height: 32,
+            marginLeft: 4,
+            padding: 8
+          }}
+        />
+      </View>
+    );
+  }
+}
+
+const LEFT_RIGHT_FLEX_WIDTH = 70;
+export default function AccountDetails({
+  account,
+  prependTransactions,
+  transactions,
+  accounts,
+  categories,
+  payees,
+  balance,
+  isNewTransaction,
+  onLoadMore,
+  onSearch,
+  onSelectTransaction
+  // refreshControl
+}) {
+  let allTransactions = useMemo(() => {
+    return prependTransactions.concat(transactions);
+  }, [prependTransactions, transactions]);
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏this `useMemo` usage is an unnecessary mico-optimization. Arguably this will actually make the performance worse.

Reply:
Haha, I'm usually wary of unnecessary memoization in my company's codebase myself.
As copy/paste, I didn't plan on changing anything so that I wouldn't need to verify it still works as before.

I'd want to verify `PagedQuery` is not maintaining referential equality for the `transactions` prop passed in here, because that would foil my usual "arrays won't be shallowly equal" instinct.
Thoughts? Maybe you're more certain than I am which way this one would go, perf-wise.

Reply:
Generally we should not be using `useMemo` for inexpensive operations. This helps keep the code clean and avoids common micro-optimization pitfalls that often actually lead to a worse performance (i.e. memoizing a value takes more time than just applying an operation on top of it; string concatenation is a prime example for this). What I always tell my teammates is: start simple and add complexity ONLY if you have a good reason for it. Otherwise: prefer KISS.

You can read more details about it in this blogpost: https://blog.logrocket.com/rethinking-hooks-memoization/

Reply:
Trust me, we don't disagree in general--I've shared that exact blog post with my own team. 
I'm just not eager to mess with working code that has no tests. (This is the [React Native AccountDetails](https://github.com/actualbudget/actual/blob/master/packages/loot-design/src/components/mobile/account.js#L50) ported over.)